src/Symfony/Bundle/TwigBundle/DependencyInjection/Configuration.php

```diff
@@ -127,7 +127,13 @@ private function addTwigOptions(ArrayNodeDefinition $rootNode)
                 ->scalarNode('cache')->defaultValue('%kernel.cache_dir%/twig')->end()
                 ->scalarNode('charset')->defaultValue('%kernel.charset%')->end()
                 ->booleanNode('debug')->defaultValue('%kernel.debug%')->end()
-                ->booleanNode('strict_variables')->end()
+                ->booleanNode('strict_variables')
+                    ->defaultValue(function () {
+                        @trigger_error('Not setting the "strict_variables" Twig configuration option to false is deprecated as of Symfony 4.1, it will be the value of the "kernel.debug" parameter by default in 5.0.', E_USER_DEPRECATED);
```

Comment:
IIRC we use to include the full path of the config option like this:

> Not setting the "twig.strict_variables" option to false [...]

Reply:
By the way, mentioning `false` here is IMO wrong. What is deprecated is not setting the option at all. So it should be:

> Not setting the "twig.strict_variables" option is deprecated [...]

Reply:
What about:
> Not setting the "strict_variables" option to "false" under "twig" configuration is deprecated as of Symfony 4.1 ...

> By the way, mentioning `false` here is IMO wrong. What is deprecated is not setting the option at all.

Yep, it could be a bit confusing, but IIUC not setting the option at all still will be possible in 5.0. We're deprecating the default value/behaviour caused by "not setting the option explicitly" at the same time we're saying to the people (the "false" part) what they need to do to keep the current behaviour in 5.0 and remove the deprecation notice during 4.1+

Reply:
My concern with the current wording is that it seems that you cannot set the option to `true` anymore.

Reply:
Now changing the point of view:
> Setting the "strict_variables" option to "false" by default under "twig" configuration is deprecated as of Symfony 4.1, it will be by default the value of the "kernel.debug" parameter in 5.0.

Wdyt? other suggestions?